mteb/model_meta.py

```diff
@@ -101,19 +107,37 @@ class ModelMeta(BaseModel):
     public_training_data: str | bool | None
     framework: list[FRAMEWORKS]
     reference: STR_URL | None = None
-    similarity_fn_name: DISTANCE_METRICS | None
+    similarity_fn_name: ScoringFunction | None
     use_instructions: bool | None
     training_datasets: dict[str, list[str]] | None
     adapted_from: str | None = None
     superseded_by: str | None = None
     citation: str | None = None
 
+    @field_validator("similarity_fn_name", mode="before")
+    @classmethod
+    def validate_similarity_fn_name(cls, value):
+        """Converts the similarity function name to the corresponding enum value.
```

Comment:
Love that this is backwards compatible! 

Just wanted to point out that at the moment, the target branch is v2, which means in v2, this field actually accepts both string and enum. I'd originally expect v2 to only accept enum, and this specific method would only be applied to `main` with a warning, saying that this field type will be switched to enum. But I'm okay with this as well 👍